shared/chat/blocking/block-modal/container.tsx

```diff
@@ -0,0 +1,66 @@
+import * as Container from '../../../util/container'
+import * as RouteTreeGen from '../../../actions/route-tree-gen'
+import BlockModal, {BlockType, NewBlocksMap} from '.'
+import * as UsersGen from '../../../actions/users-gen'
+import * as TeamsGen from '../../../actions/teams-gen'
+
+type OwnProps = Container.RouteProps<{
+  blockByDefault?: boolean
+  convID?: string
+  others?: Array<string>
+  team?: string
+  username: string
+}>
+
+const Connect = Container.connect(
+  (state, ownProps: OwnProps) => ({
+    adderUsername: Container.getRouteProps(ownProps, 'username', ''),
+    blockByDefault: Container.getRouteProps(ownProps, 'blockByDefault', false),
+    convID: Container.getRouteProps(ownProps, 'convID', undefined),
+    isBlocked: (username: string, which: BlockType) => {
```

Comment:
It's bad for performance to create new objects/closures in `mapStateToProps`, prefer passing down the data and creating this in `mergeProps` instead.

Reply:
do i pass down `dispatch`?

Reply:
ah nevermind this is mapStateToProps, not mapDispatchToProps, sorry!